convert2rhel/checks.py

```diff
@@ -57,6 +62,88 @@ def perform_pre_ponr_checks():
+    )
+
+    repo_dir = os.path.join(utils.TMP_DIR, "convert2rhel_repo")
+    utils.mkdir_p(repo_dir)
+    # repo_dir = os.path.join("/tmp/","convert2rhel_repo")
+    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
+    store_content_to_file(filename=repo_path, content=repo_content)
+
+    # repoquery --setopt=reposdir=$(pwd)/test_repo --releasever=8 convert2rhel
+    raw_output_convert2rhel_versions, _ = run_subprocess(
+        "repoquery --releasever={releasever} --setopt=reposdir={repodir} convert2rhel".format(
+            releasever=system_info.releasever,
+            # releasever="8",
+            repodir=repo_dir,
+        ),
+        print_output=False,
+    )
+
+    #   - if cannot access url: print warning and continue
+    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
+    convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
+
+    latest_version = ("0", "0.00", "0")
+    #   - get latest version from https://ftp.redhat.com/redhat/convert2rhel/.
+    for package_version in convert2rhel_versions:
+        if package_version[0] != "convert2rhel":
+            continue
+
+        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+
+    # if current version is not latest:
+    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
+    # - If envar is detected
+    if "CONVERT2RHEL_UNSUPPORTED_VERSION" in os.environ:
+        logger.warning(
+            "You are currently running %s, the latest version of convert2rhel is %s.\n"
+            "'CONVERT2RHEL_UNSUPPORTED_VERSION' environment detected, continuing conversion"
+            % (convert2rhel_version, latest_version)
+        )
+
+    # - If converting rhel6:
+    elif ver_compare == -1:
+        if system_info.version.major <= 6:
+            logger.critical(
+                "You are currently running %s, the latest version of convert2rhel is %s.\n"
+                "Only the latest version is supported for conversion. If you want to ignore"
+                " this you can set 'CONVERT2RHEL_UNSUPPORTED_VERSION' to continue"
+                % (convert2rhel_version, latest_version)
+            )
+
+        else:
+            logger.info(
+                "You are currently running %s, the latest version of convert2rhel is %s.\n"
+                "Continuing conversion." % (convert2rhel_version, latest_version)
+            )
+
+    # - If we are converting rhel7 or rhel8:
+    elif ver_compare == 1:
+        latest_version = package_version[1:]
+        logger.info(
+            "You are currently running %s, the latest version of convert2rhel is %s.\n"
+            "Continuing conversion." % (convert2rhel_version, latest_version)
+        )
+
+    else:
+        logger.info(
+            "You are currently running %s, the latest version of convert2rhel is %s.\n"
+            "Continuing conversion." % (convert2rhel_version, latest_version)
+        )
```

Comment:
I'm wondering, @Andrew-ang9, should this code all be inside the for-loop? Right now, it's outside of the loop and we are comparing the current convert2rhel version against the ("0", "0.0", "0") tuple in `latest_version`

Reply:
I think @rodolfo is partially right here.  I notice that you are setting `ver_compare` from `rpm.labelCompare()` both inside of the loop and outside of the loop.  I think what may have happened is you needed to do something like this pseudo-code inside of the loop:
``` python
for package_version in packages:
    ver_compare = rpm.labelCompare(package_version, latest_version)
    if ver_compare > 0:
        latest_version = package_version
```

and then you needed to do something like this pseudo-code after the loop:
``` python
ver_compare = rpm.labelCompare(running_version, latest_version)
if ver_compare <= 0:
    # Do all the error checking
```
But when you saw you were checking ver_comp again, you merged the conditions inside the loop with the conditions outside of the loop (which is how `latest_version = package_version[1:]` ended up on line 134.)

In this case, what you have to remember, is even though ver_compare is the same variable name inside and outside of the loop, what it contains is different.  Inside of the loop, it is a comparison between another package version and the latest package version found so far. (your iterating through each package version, doing the comaprison, trying to find the latest package version that exists).  Outside of the loop, it is a comparison between the latest package version and the running version.

Reply:
I have went through and changed this code!